testing/python/collect.py

```diff
@@ -841,6 +841,33 @@ def test_makeitem_non_underscore(self, testdir, monkeypatch):
         l = modcol.collect()
         assert '_hello' not in l
 
+    def test_issue2369_collect_module_fileext(self, testdir):
+        """Ensure we can collect files with weird file extensions as Python
+        modules (#2369)"""
+        # We'll implement a little finder and loader to import files containing
+        # Python source code whose file extension is ".narf".
+        testdir.makeconftest("""
+            import sys, os, imp
+            from _pytest.python import Module
+
+            class Loader:
```

Comment:
Clever solution. 😁 

I notice though that you are using `sys.meta_path.append`, and because we run this test in the same process as the main process running pytest, we are leaving that finder there for the rest of pytest's test run.

The simplest solution I can find now is to just use `testdir.runpytest_subprocess()` to run that test in a separated process to ensure isolation. What do you think?